packages/fxa-content-server/app/scripts/models/account.js

```diff
@@ -372,6 +382,127 @@ const Account = Backbone.Model.extend(
       });
     },
 
+    _fetchAccountProfileResult: null,
+
+    /**
+     * Fetches the account profile from the GET /account/profile on the auth server and caches the
+     * result locally in _accountProfileFetchPromise. This also updates the account model
+     * with the new properties (minus any verificationMethod, verificationReason values)
+     *
+     * @returns {Promise}
+     */
+    async fetchAccountProfile() {
+      if (this._fetchAccountProfileResult) {
+        return this._fetchAccountProfileResult;
+      }
+
+      const result = await this.accountProfile();
+
+      // Currently, we only populate the `ecosystemAnonId`, however we could examine populating
+      // other account values.
+      if (result.ecosystemAnonId) {
+        this.set('ecosystemAnonId', result.ecosystemAnonId);
+      }
+
+      this._fetchAccountProfileResult = result;
+
+      return result;
+    },
+
+    /**
+     * Attempt to generate and set the ecosystem anon if it isn't already set. At a high level
+     * this function attempts to generate the id in the following manner:
+     *
+     * 1. Use kB if it is specified
+     * 2. Attempt to fetch keys if valid authentication token exists to do so
+     * 3. Use user's password to fetch keys
+     *
+     *
+     * @param {Object} [options={}]
+     *   @param {String} [options.kB] Optional The user's kB, ff specified takes priority.
+     *   @param {String} [options.password] Optional The user's password
+     * @returns {Promise}
+     */
+    async generateEcosystemAnonIdIfRequired(options = {}) {
+      try {
+        let ecosystemAnonId = this.get('ecosystemAnonId');
+        if (ecosystemAnonId) {
+          return;
+        }
```

Comment:
If I'm understanding that Dave will be using this method in various views, do we want to allow for updating the ID when the user changes their password, in which case I think this could be `if (ecosystemAnonId && !options.kB)`?

Reply:
Hrm, I wonder what value the `IfRequired` part gives us now. I'm going to drop that part and remove this check since we can't guarantee saving it.